### PR TITLE
Update Kokkos library in LAMMPS to v4.4.1

### PR DESCRIPTION
**Pull request by stanmoore1**
**Summary**

- Update Kokkos library in LAMMPS to v4.4.1
- Disable CUDA malloc async which causes crashes with some MPI versions
- Add `-DKokkos_ENABLE_ATOMICS_BYPASS=ON` for Serial builds

**Related Issue(s)**

None

**Author(s)**

Stan Moore (SNL) and Kokkos developers

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

Yes

### CODE REVIEW

File: cmake/Modules/Packages/KOKKOS.cmake [hunk 1]
```diff
@@ -10,6 +10,21 @@ endif()
 if(Kokkos_ENABLE_CUDA)
   message(STATUS "KOKKOS: Enabling CUDA LAMBDA function support")
   set(Kokkos_ENABLE_CUDA_LAMBDA ON CACHE BOOL "" FORCE)
```
Comment (rbberger) [inline cmake/Modules/Packages/KOKKOS.cmake]: `Kokkos_ENABLE_CUDA_LAMBDA` is no longer an option since `v4.1.0`

Comment (stanmoore1) [reply cmake/Modules/Packages/KOKKOS.cmake]: Removed

**Comment by stanmoore1** [comment]
@akohlmey I think [2429c89](https://github.com/lammps/lammps/pull/4322/commits/2429c89eae2d6f680c3b454911f4da9e76f1fadd) and [97627bd](https://github.com/lammps/lammps/pull/4322/commits/97627bd77ab4adaec43e714c2a341e90c9b25204) need to go into the next stable update as bugfixes

**Comment by akohlmey** [comment]
> @akohlmey I think [2429c89](https://github.com/lammps/lammps/pull/4322/commits/2429c89eae2d6f680c3b454911f4da9e76f1fadd) and [97627bd](https://github.com/lammps/lammps/pull/4322/commits/97627bd77ab4adaec43e714c2a341e90c9b25204) need to go into the next stable update as bugfixes

Cherry-picked the first, the second is already there as commit 89d3c0b0a6bf8b597166565fd410fbac3f1cf0ae

**Comment by stanmoore1** [comment]
@akohlmey thanks!

**Comment by stanmoore1** [comment]
No new regression test failures.

**Comment by stanmoore1** [comment]
Performance on V100 looks unchanged.

**Comment by stanmoore1** [comment]
I tested the new CMake options with HIP and CUDA and all seems to be working. This is ready to merge from my POV.

**Review by sjplimp — APPROVED** [review]

**Pull request merged**